src/core.js

```diff
@@ -300,15 +300,13 @@ jQuery.extend({
 	makeArray: function( arr, results ) {
 		var ret = results || [];
 
-		if ( arr != null ) {
-			if ( isArraylike( Object(arr) ) ) {
-				jQuery.merge( ret,
-					typeof arr === "string" ?
-					[ arr ] : arr
-				);
-			} else {
-				push.call( ret, arr );
-			}
+		if ( isArraylike( Object(arr) ) ) {
+			jQuery.merge( ret,
+				typeof arr === "string" ?
+				[ arr ] : arr
+			);
+		} else if ( arr != null ) {
+			push.call( ret, arr );
```

Comment:
Oh right, it's because yesterday I tried to remove the call of `Object(...)` but it failed with `"string"` after :/